### DISCUSSION TITLE
The training process is frozen

### DISCUSSION THREAD

**Discussion by sangnguyen1006** [ultralytics/ultralytics#7788] [Q&A]
I've only recently used the yolo model and don't have much experience yet. I'm training the yolov8.1 detection model on a custom dataset, everything was fine until the start of training, it seems like my data file wasn't loaded for training. No error message and the training process freezes here. Can anyone tell me what I should do in such a case? Any suggestions would be very important to me. Thank you!

     active environment : ultralytics_env
    active env location : /home/sangnguyen/.conda/envs/ultralytics_env
            shell level : 2
       user config file : /home/sangnguyen/.condarcpopulated config files : /home/sangnguyen/.condarc
          conda version : 23.11.0
    conda-build version : not installed
         python version : 3.11.5.final.0
                 solver : libmamba (default)
       virtual packages : __archspec=1=zen4
                          __conda=23.11.0=0
                          __glibc=2.31=0
                          __linux=5.14.21=0
                          __unix=0=0
       base environment : /usr/miniconda3  (read only)
      conda av data dir : /usr/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://conda.anaconda.org/pytorch/linux-64
                          https://conda.anaconda.org/pytorch/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://conda.anaconda.org/bioconda/linux-64
                          https://conda.anaconda.org/bioconda/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /usr/miniconda3/pkgs
                          /home/sangnguyen/.conda/pkgs
       envs directories : /home/sangnguyen/.conda/envs
                          /usr/miniconda3/envs
               platform : linux-64
             user-agent : conda/23.11.0 requests/2.28.1 CPython/3.11.5 Linux/5.14.21-150400.22-default sles/15.4 glibc/2.31 solver/libmamba conda-libmamba-solver/23.12.0 libmambapy/1.5.3
             netrc file : None
           offline mode : False

Ultralytics YOLOv8.1.5 🚀 Python-3.11.7 torch-2.1.2 CPU (AMD EPYC 9454 48-Core Processor)

in python: model.train(data = 'YAML_PATH', epochs = 2, batch = 16, imgsz = 1280, device = 'cpu', workers = 16)

![Untitled](https://github.com/ultralytics/ultralytics/assets/60694260/af7fbcb7-61ad-4e47-9692-8b748b33723e)

**Comment by pderrenger** [comment]
@sangnguyen1006 it seems like you're encountering an issue where the training process is not proceeding as expected. Here are a few steps you can take to troubleshoot the problem:

1. **Check YAML File**: Ensure that your dataset YAML file is correctly formatted and all paths to your training, validation images, and labels are correct. The dataset should be structured properly as per the documentation.

2. **Verify Dataset**: Make sure that your custom dataset is accessible and the images/annotations are correctly loaded. You can use the `val` mode to run a quick check on your dataset without training.

3. **Environment Check**: Confirm that your Python environment has all the necessary dependencies installed and there are no conflicts. Since you're using a conda environment, ensure it's activated and all packages are compatible.

4. **Hardware Resources**: Although you're using the CPU for training, check if your system has enough memory and processing power to handle the dataset and model size you're working with.

5. **Logs and Outputs**: Look for any logs or console outputs that might give you a hint about where the process is getting stuck. If there are no outputs, consider adding print statements in your training script to debug and find the step where it freezes.

6. **Update YOLOv8**: Ensure you're using the latest version of YOLOv8 as there might be bug fixes and improvements that could resolve your issue.

7. **Minimal Example**: Try running a minimal training example with a very small dataset to see if the issue persists. This can help isolate whether the problem is with the dataset or the training setup.

8. **Check for Errors**: Although you mentioned there are no error messages, it's worth double-checking the terminal or command prompt for any warnings or errors that might have been overlooked.

If after trying these steps you're still facing issues, you can reach out to the community with specific details about the steps you've taken, and any logs or error messages you've encountered. This will help in diagnosing the problem more effectively. Remember to keep your training script and dataset as simple as possible while troubleshooting to eliminate any potential complexities.
